locals.tf

```diff
@@ -1,14 +1,16 @@
 locals {
   # automatic upgrades are either:
-  # - null 
-  # - patch, but then the kubernetes_version musnt't specify a patch number
-  # - rapid/stable/node-image, but then the kubernetes_version must be null
+  # - null
+  # - patch, but then the kubernetes_version must not specify a patch number and orchestrator_version must be null
+  # - rapid/stable/node-image, but then the kubernetes_version and the orchestrator_version must be null
   automatic_channel_upgrade_check = var.automatic_channel_upgrade == null ? true : (
     contains(["patch"], var.automatic_channel_upgrade) &&
-    can(regex("^[0-9]{1,}\\.[0-9]{1,}$", var.kubernetes_version))
+    can(regex("^[0-9]{1,}\\.[0-9]{1,}$", var.kubernetes_version)) &&
+    var.orchestrator_version == null
     ) ? true : (
     contains(["rapid", "stable", "node-image"], var.automatic_channel_upgrade) &&
-    var.kubernetes_version == null
+    var.kubernetes_version == null &&
+    var.orchestrator_version == null
   )
```

Comment:
I would like to try:

```hcl
automatic_channel_upgrade_check = var.automatic_channel_upgrade == null ? true : var.orchestrator_version == null && (
    (contains(["patch"], var.automatic_channel_upgrade) && can(regex("^[0-9]{1,}\\.[0-9]{1,}$", var.kubernetes_version)))
    || (contains(["rapid", "stable", "node-image"], var.automatic_channel_upgrade) && var.kubernetes_version == null
  ))
```

Reply:
Done ! Implemented new unit tests to test both when it should return automatic_channel_upgrade_check `true` and automatic_channel_upgrade_check `false`